drivers/hmis/app/graphql/mutations/submit_assessment.rb

```diff
@@ -8,34 +8,51 @@ class SubmitAssessment < BaseMutation
 
     def resolve(input:)
       assessment, errors = input.find_or_create_assessment
-      return { assessment: nil, errors: errors } if errors.any?
+      return { errors: errors } if errors.any?
 
       definition = assessment.assessment_detail.definition
       enrollment = assessment.enrollment
 
-      # Determine the Assessment Date (same as Information Date) and validate it
+      # If this is an HoH Exit, check special restrictions
+      new_hoh_enrollment = nil
+      if enrollment.head_of_household? && assessment.exit?
+        open_enrollments = Hmis::Hud::Enrollment.open_on_date.
+          viewable_by(current_user).
```

Comment:
Maybe you should not check viewability here? -- if it is ever be the case that there might be open enrollments in a household that the current user can't see, then this might allow for exiting the HoH without moving the HoH?

The error doesn't expose any PII, so I think it is safe to note that there are other enrollments. If we think users would be confused, maybe make the message identify this case (e.g., "... open enrollments _that you cannot see_. ")?

Reply:
That was my original thought, but @eanders confirmed that if you can see _one_ member of the household, you can always see all of them. Do you think otherwise?

The purpose of using viewable_by here is actually to apply the filtering by data_source_id, since we can't rely on Household ID to be unique.